Improvements: Summary Addition, Keyword Configuration Separation, Secret Masking, and Resource Protection (#27)
wso2-extensions/identity-tools-cli: docs/cli-mode.md

```
> **Note:** Before running the CLI commands, be sure to export the environment variables with the correct server details that the action should be performed against.
> 
> It is recommended to use the ```serverConfig.json``` file to provide the server configurations as it is more secure and easier to maintain when dealing with multiple environments.

#### Using environment variables in serverConfig.json
You can also explicitly specify the use of environment variables for certain configurations in the ```serverConfig.json``` file itself. To do this, use the placeholder ```${YOUR_ENV_VAR_NAME}``` in the ```serverConfig.json``` file, as shown in the following example:
```
{
  "CLIENT_ID": "${DEV_CLIENT_ID}",
  "CLIENT_SECRET": "${DEV_CLIENT_SECRET}",
  "SERVER_URL": "https://localhost:9443",
  "TENANT_DOMAIN": "carbon.super"
}
```
The tool will search for the keyword with the name given inside the placeholder in the environment and use its value instead.

### Tool configurations
The ```toolConfig.json``` file contains the configurations needed for overriding the default behaviour of the tool. 

Example configuration file:
```
{
   "ALLOW_DELETE" : true,

>    "EXCLUDE" : ["Claims"]

Reviewer: Let's add additional configs to represent userstores and claims as well for examples.

   }
}
```
### Exclude or Include Specific Resource Types
In addition to excluding or including specific resources by name, it is also possible to exclude or include entire resource types. 

To exclude specific resource types:
```
{
    "EXCLUDE": ["ResourceType1", "ResourceType2"]
}
```
To include only selected resource types:
```
{
    "INCLUDE_ONLY": ["ResourceType1", "ResourceType2"]
}
```
Example:
```
{
   "EXCLUDE": ["IdentityProviders", "UserStores"]
}
```
> **Note:** When both EXCLUDE and INCLUDE_ONLY properties are used, INCLUDE_ONLY takes precedence over EXCLUDE.

#### Exclude secrets from exported resources

> By default, secrets are removed from the exported resources. For applications, identity providers, and user stores the value of secrets will be masked by a string: ```'********'```.

Reviewer: ```suggestion
By default, secrets fields are masked by a string: ```'********'```.
```
